File: builders/testdata/nodejs/functions/local_dependency_yarn/function.js
/**
 * Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * Responds 'PASS' if the 'server' local dependency was installed correctly.
 *
 * @param {!Object} req request context.
 * @param {!Object} res response context.
 */
exports.testFunction = (req, res) => {
  const pkg = require('server/package.json');
  const got = pkg.version;
  const want = '0.5.2';
  if (got == want) {
    res.send('PASS');
  } else {
    res.send(`Unexpected server version: got ${got}, want ${want}`);
  }
};

File: builders/testdata/go/functions/non_declarative_cloud_event/function.go
// Copyright 2021 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

// Package fn tests a non-declarative CloudEvent function.
package fn

import (
	"context"
	"errors"

	cloudevents "github.com/cloudevents/sdk-go/v2"
)

// Func is a non-declarative CloudEvent function.
func Func(ctx context.Context, e cloudevents.Event) error {
	var data string
	if err := e.DataAs(&data); err != nil || len(data) == 0 {
		return errors.New("FAIL")
	}

	return nil
}
